proposals/blittable.md

```diff
@@ -8,129 +8,92 @@
-
-## Detailed design
-[design]: #detailed-design
+Type parameters with the unmanaged constraint can use all the features available to unmanaged types: pointers, fixed, etc ... 
 
-The language will introduce a new declaration modifier named `blittable` that can be applied to `struct` definitions.  
 
 ``` c#
-blittable struct Point 
+void Hash<T>(T value) where T : unmanaged
 {
-    public int X;
-    public int Y;
+    // Okay
+    using (T* p = &value) { 
+        ...
+    }
 }
 ```
 
-The compiler will enforce that the fields of such a struct definition fit one of the following categories:
-
-- `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `char`, `float`, `double`, `decimal`, or `bool`.
-- Any `enum` type
-- Any pointer type which points to a `blittable` type
-- Any user defined struct explicitly declared as `blittable`
-
-Any type fitting the definition above, or any element in the list, is considered a valid `blittable` type.
+This constraint will also make it possible to have efficient conversions between structured data and streams of bytes. This is an operation that is common in networking stacks and serialization layers:
 
 ``` c#
-struct User
-{
-    string FirstName;
-    string LastName;
-}
-
-blittable struct ItemData
-{
-    // Error: blittable struct cannot contain field of type User which is not blittable
-    User User;
-    int Id;
+Span<byte> Convert<T>(ref T value) where T : unmanaged {
+    ...
 }
 ```
 
-Note that a user defined struct must be explicitly declared as `blittable` in order to meet the requirements above.  This is required even if the struct otherwise meets the requirements of `blittable`. 
+Such routines are advantageous because they are provably safe at compile time and allocation free.  Interop authors today can not do this (even though it's at a layer where perf is critical).  Instead they need to rely on allocating routines that have expensive runtime checks to verify values are correctly unmanaged.
 
-``` c#
-struct SimplePoint
-{
-    public int X;
-    public int Y;
-}
+## Detailed design
+[design]: #detailed-design
 
-blittable struct Data
-{
-    // Error: blittable struct cannot contain field of type SimplePoint which is not blittable
-    SimplePoint Point;
-}
-```
+The language will introduce a new constraint named `unmanaged`. In order to satisfy this constraint a type must be a struct and all the fields of the type must fall into one of the following categories:
 
-The language will also support the ability to constrain generic type parameters to be `blittable` types.  
+- Have the type `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `char`, `float`, `double`, `decimal`, or `bool`.
```

Comment:
Is it worth explicitly calling out `IntPtr` and `UIntPtr`? These are primitives from the perspective of the runtime.

What about explicitly sized structs or structs with no fields?

Reply:
> Is it worth explicitly calling out IntPtr and UIntPtr? 

I will add `IntPtr` and `UIntPtr`. 

> What about explicitly sized structs or structs with no fields?

What do you think is special about them? The lack of fields mean they implicitly meet the requirements above.

Reply:
Was asking because the wording basically must have the type or be comprised of the types, lack of fields is kind of vague, so I was wondering if it needed an explicit callout